src/dmd/json.d

```diff
@@ -793,19 +793,62 @@ public:
         jsonProperties(d);
         objectEnd();
     }
+
+    private void generateBuildInfo()
+    {
+        objectStart();
+        property("kind", "buildInfo");
+        property("binary", global.params.argv0);
+        property("version", global._version);
+        property("config", global.inifilename ? global.inifilename : null);
+        if (global.params.lib) {
+            property("library", global.params.libname);
+        }
+        objectEnd();
+    }
+    private void generateSemantics()
+    {
+        objectStart();
+        property("kind", "semantics");
+        propertyStart("modules");
+        arrayStart();
+        foreach (m; Module.amodules)
+        {
+            objectStart();
+            if(m.md)
+                property("name", m.md.toChars());
+            property("file", m.srcfile.toChars());
```

Comment:
how about `absolutePath ( m.srcfile.toChars() )` so that if rdmd is called from a different directory, it'll still work?

Reply:
hmmmm, interesting thought.  I'd have to think about that one.  This is mimicking the current functionality from the verbose output, but maybe it can be improved.  I'll come back to this.

Reply:
NOTE: "contentImports" appears to use absolutePath (i tried with -J. and `import("foo.txt")` ), so I would make sense to be consistent. I suggest using absolutePath for `file` as well (and wherever else it makes sense)

Reply:
I believe that would be changing behavior though...we could end up with a regression if we change the behavior.

Reply:
well, that info wasn't in json output before... so it's one of those gray area things.
one option is to allow configuring json output (as I suggested somewhere else), which is OK if we have good way to customize, eg:

`-Xf=file -Xopt=abspaths,-functions`

Reply:
What I mean is that `rdmd` will be using that field the same way it was using the `import   ` line from the verbose output.  And in that case it was using the same `m.srcfile.toChars()` expression.  If we use the absolute path instead moving forward, I just want to make sure it won't break anything.

Reply:
Now that I think about it, there might be a bug in the current design.  If you build with rdmd, and then you change directories and build again, then all the relative module filenames will be incorrect.  It might just force you to do a rebuild, but that's no good.  We could fix this using absolute paths, or we could save off the original CWD that rdmd was using when it first performed the build.

Reply:
There should be a test for that in the rdmd test suite, FWIW.

Reply:
there may even be corner cases where it picks up wrong file eg:

ls
main.d
fun.d
temp/fun.d // slightly different version main.d

rdmd main.d
cd temp
rdmd ../main.d
=> can it get confused and get the wrong fun.d?

Here's a good compromise:

add a CWD field, and when rdmd reads a path (any path) call  buildPath(json.CWD, json.path) (does the right thing w absolute paths)

and allow for:
rdmd main.d
cd temp
rdmd ../main.d => should reuse cache and not rebuild

Reply:
but we should definitely make behavior consistent between contentImports and modules.file:
import("fun.txt") should not get auto-transformed in an absolute path either.

Reply:
I'm thinking that the JSON file saves the CWD in the new `buildInfo` object.